### PR TITLE
Fix for more complex inline comment examples

### PR DESCRIPTION
**Pull request by kipz**
Related to #11 but fixes some more complex examples of inline comments found in the wild

### CODE REVIEW

File: parser.js [hunk 1]
```diff
@@ -382,9 +381,10 @@ function parse(contents, options) {
...
         var nextLine = lines[i];
         // remove inline RUN # comment see https://github.com/joyent/node-docker-file-parser/issues/8
-        if (nextLine.match(TOKEN_INLINE_COMMENT)) {
-            nextLine = "\\";
-        }
+        // if line will be continued, comments don't matter :=
+        if (line && line.match(TOKEN_LINE_CONTINUATION) && isComment(nextLine)) {
+            line = remainder + "\\";
+        } else
         if (remainder) {
```
Comment (twhiteman) [inline parser.js:388]:
```suggestion
        } else if (remainder) {
```

File: test/test_parser.js [hunk 2]
```diff
@@ -193,3 +193,28 @@ tape('inline comments', function (t) {
...
+  var dockerFile = fs.readFileSync(dname + '/ComplexDockerfile', 'utf8');
+  var commands = dockerfileParser.parse(dockerFile);
+  t.equal(commands[0].lineno, 1,  'line number should be 1');
+  t.equal(commands[1].lineno, 4,  'line number should be 4');
+  t.equal(commands[2].lineno, 6,  'line number should be 6');
+  t.equal(commands[3].lineno, 56, 'line number should be 56');
+  t.equal(commands[4].lineno, 58, 'line number should be 58');
+  t.equal(commands[5].lineno, 89, 'line number should be 89');
+  t.end();
+});
```
Comment (twhiteman) [inline test/test_parser.js:220]: newline here would be appreciated

File: test/test_parser.js [hunk 3]
```diff
@@ -193,3 +193,28 @@ tape('inline comments', function (t) {
...
 });
+
+tape('line numbers for files without whitespace before inline comments', function (t) {
+
+  var dname = __dirname;
+  var dockerFile = fs.readFileSync(dname + '/Dockerfile-with-inline-comment-no-ws', 'utf8');
+  var commands = dockerfileParser.parse(dockerFile);
+
+  t.equal(commands[0].lineno, 1,  'Line number should be 1');
+  t.equal(commands[1].lineno, 5,  'Line number should be 5');
```
Comment (twhiteman) [inline test/test_parser.js:204]: Would be also good to check the `command[1].args` is correct (i.e. the comment was stripped).

**Review by twhiteman — CHANGES_REQUESTED** [review]
Looks good - would appreciate a few little tweaks before pushing through - thanks!

**Comment by kipz** [comment]
Thanks for reviewing @twhiteman

**Review by twhiteman — APPROVED** [review]
LGTM - thanks for the PR.

**Pull request merged**